src/data/calculations/CalculationsRepository.ts

```diff
@@ -19,40 +23,35 @@ export class CalculationsRepository implements ICalculationsRepository {
             ongoingCalculation: ongoing,
         });
     }
-    async storeCalculation(
-        calculationData: CalculationDataType
-    ): Promise<void> {
+    async storeCalculation(calculationData: CalculationData): Promise<void> {
         const oldCalculations = await this.getAllCalculations();
         const newCalculations = [calculationData, ...oldCalculations];
         await this.remoteDataSource.set({
-            allCalculations: JSON.stringify(newCalculations),
+            allCalculations: calculationDataArrayToJSON(newCalculations),
         });
     }
 
-    async getAllCalculations(): Promise<CalculationDataType[]> {
+    async getAllCalculations(): Promise<CalculationData[]> {
```

Comment:
interesting you guys didnt opt for using arrow functions - is there a reason for the preference in convention ?

Reply:
It's my syntax habits 😵‍💫 I can change them all, do you use a linter to enforce the style?

Reply:
no purely preference just more used to seeing arrow functions

Reply:
- I've already used arrow functions elsewhere, just forgot to use them when writing the classes as well.
- Just did a check on regular vs arrow functions in JS classes, and there is actually a difference in behaviour:
  - instance methods that access instance variables 
    - cannot be reassigned if they are regular functions
    - can be reassigned if they are arrow functions

[Playground](https://www.typescriptlang.org/play?target=6#code/MYGwhgzhAEAqCmEAuBhcVoG8BQ0-QgHsBbeABQCdCAHaAXmgHIAJASwEJGBubXfagK4AjEK2DQwFKgHcAYgIB2wJK0IL60ABQBKAFwEkFVgoDm9AHxY++fBXhIBFdQCIAko2ISphaQDpoztAA1NBIABasEL5EpJQ0PDYAvrw2giJi0HYmAuAU8koqajrQ+shGplY2NnYOTgHunlk5kv6BIeGR0STkVNQJ+MnYg8BqRCDwviCEJprOKGAgohUAZlSexshgSvDO2rwjCsihiKjoMAwK8NJwJ2iQEDo8B2MTUzNIt2e+kjL5yqoKHTaJ6jQjjSbTTQfZB3KC+Jq5P6FQHaYH7UHgt6zeaLYxmVYkTLwe6sEyXAAmu3RhyQEigpIUACV4NlEYp-moNNDTvd4Szmnl2ciQTS6RAGQBBbxyIUArmfXk-HxIgEil4QmYkslS36yoqo7DPMGvSFapn8tkFAFA3hAA) 
![image](https://github.com/Theodo-UK/sustainability-calculator/assets/57725347/0a693cfa-1190-4c34-91b1-2cf669201747)

Reply:
I will leave this as a regular function then, since the methods in Repositories are not meant to be reassigned!